aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -735,6 +735,17 @@ template<> void blasTriangularSolve<float>(char side, char uplo, char trans, cha
 }
 #endif
 
+void _linalg_check_errors(
+    const Tensor& info,
+    const c10::string_view api_name,
+    bool is_matrix) {
+  if (is_matrix) {
+    singleCheckErrors(info.item<int64_t>(), api_name);
+  } else {
+    batchCheckErrors(info, api_name);
+  }
+}
```

Comment:
On second thought... does `is_matrix` need to be an argument to this function? Could we infer `is_matrix` from `info.numel() == 1`? cc @Lezcano @IvanYashchuk

Reply:
We can't do that because matrices of size `(1, 1, 3, 2)` and `(3, 2)` would be treated the same.
Now, we could probably infer from `info.dim() == 0`. What do you think @IvanYashchuk ?

Reply:
I don't know if `info` is consistently implemented to have shape `()` for single input matrices, it could also be `(1,)`.

Reply:
@Lezcano @IvanYashchuk another question: is the condition always "is a matrix"? Looking at some code (I decided to change usages of batchCheckErrors to this new logic in this PR to avoid churn), there are some cases where the condition is "result.dim() > 1":

https://github.com/pytorch/pytorch/blob/9aa1b3e396c1fe9dadecaee04c9321f659d863ac/aten/src/ATen/native/BatchLinearAlgebra.cpp#L962-L976

Should I rename (and invert) `is_matrix` to `is_batch` instead?

Reply:
@zou3519, thank you for pointing out this use case that deviates from other uses of `info` to check the error! It was necessary here because input could be a single matrix but then internally it was expanded to be batched which affects the size of `infos`.

I submitted a pull request (https://github.com/pytorch/pytorch/pull/69752) fixing that and now the condition is `if (input.dim() > 2)` instead of `if (vector_case ? result.dim() > 1 : result.dim() > 2)`.

> Should I rename (and invert) is_matrix to is_batch instead?

I don't think it's necessary, only if you prefer "is_batch".